src/modules/Footer.jsx

```diff
@@ -0,0 +1,269 @@
+/* eslint-disable react/prop-types */
+import React, { useEffect, useState } from 'react';
+import ArrowBackIcon from '@material-ui/icons/ArrowBack';
+import ArrowForwardIcon from '@material-ui/icons/ArrowForward';
+import ArrowDownwardIcon from '@material-ui/icons/ArrowDownward';
+import ArrowUpwardIcon from '@material-ui/icons/ArrowUpward';
+import SlowMotionVideoIcon from '@material-ui/icons/SlowMotionVideo';
+import { getNewBoardArray, areArraysEqual, setValueInLocalStorage } from './utils/helpers';
+
+const Footer = (props) => {
+  const {
+    bestScore,
+    setBestScore,
+    boardSize,
+    history,
+    setHistory,
+    setOpenWin,
+    setOpenLose,
+    winningNumber,
+  } = props;
+  const currentBestScore = bestScore[bestScore.length - 1] ? bestScore[bestScore.length - 1] : 0;
+  const current = history[history.length - 1];
+  const { board, score, win, gameOver } = current;
+  const squareBoardSize = boardSize * boardSize;
+  let newScore = 0;
+  const [play, setPlay] = useState(false);
+
+  const isWin = (array) => {
+    const tempArr = [...array].sort((a, b) => a - b);
+    return tempArr[tempArr.length - 1] >= winningNumber;
+  };
+
+  const handleGameOver = () => {
+    if (!gameOver) {
+      setOpenLose(true);
+      setPlay(false);
+    }
+
+    return true;
+  };
+
+  const setNewBoard = (newBoard) => {
+    const checkObj = {
+      resultBoard: null,
+      winCheck: null,
+      gameOver: null,
+    };
+
+    if (!areArraysEqual(newBoard, board)) {
+      const tempBoard = getNewBoardArray(newBoard);
+      checkObj.resultBoard = tempBoard !== -1 ? tempBoard : newBoard;
+      if (isWin(checkObj.resultBoard) && !win) {
+        checkObj.winCheck = true;
+        setOpenWin(true);
+      }
+      // eslint-disable-next-line no-use-before-define
```

Comment:
it's a valuable suggestion, remember about hoisting?

Reply:
yep. I'll try to avoid using before declaration.

Reply:
It can work ok if you’d used function declarations instead of function expressions.

Reply:
damn, I've totally forgotten about hoisting of function declarations, now your question makes more sense))